### ISSUE TITLE
Change Request: Can we make the Linter.Settings an interface for module declaration merging?

### ISSUE THREAD

**Issue by MWhite-22** [eslint/eslint#19662]
### ESLint version

9.25.1

### What problem do you want to solve?

Settings is current typed as any record `{ [key: string]: unknown }`, which is totally valid. However, some plugins make use of the settings options pretty extensively (like react, jsx-a11y, perfectionist, etc.).

If the Linter.Settings was an interface type, then every consumer could optionally add a module declaration to the package, and add the plugin's type info into the setting.

Example:

```ts
declare module 'eslint' {
    interface ConfigSettings {
        'jsx-a11y': {
            components: Record<string, string>;
        };
    }
}

export default defineConfig({
    settings: {
        // ...IDE Now gives me type suggestions for a `jsx-a11y` property in here!
    }
});

```

### What do you think is the correct solution?

It keeps all the current functionality and is the TS recommended way of letting consumers add their own type info at their own discretion.

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
I think this is a good idea and we can look at doing this after we move the config types into `@eslint/core`. I don't want to add more types in the `eslint` package that will make it more difficult when we extract these types.

**Comment by fasttime** [comment]
This isn't stale. It's blocked on a pull request to review in `eslint/rewrite`: https://github.com/eslint/rewrite/issues/208.

**Comment by nzakas** [comment]
I'm working on this.

**Comment by lumirlumir** [comment]
Marking this issue as accepted, as there is an ongoing PR addressing it: https://github.com/eslint/eslint/pull/19843
